pkg/controller/replica.go

```diff
@@ -132,20 +133,22 @@ func (r *ReconcileReplica) Reconcile(ctx context.Context, request reconcile.Requ
 
 func (r *ReconcileReplica) triggerGarbageCollection(volumeName string) {
 	emptyCtx := context.TODO()
-	deletionCtx, cancelFunc := context.WithCancel(emptyCtx)
-	if _, ok := r.cleanUpMap.LoadOrStore(volumeName, cancelFunc); ok {
+
+	if _, ok := r.cleanUpMap.Load(volumeName); ok {
 		klog.Infof("There already is a scheduled garbage collection for AzVolume (%s)")
-		cancelFunc()
 		return
 	}
+	deletionCtx, cancelFunc := context.WithCancel(emptyCtx)
+	now := time.Now()
+	r.cleanUpMap.Store(volumeName, cleanUpEntry{ctx: deletionCtx, cancelFunc: cancelFunc, initTimestamp: now})
```

Comment:
You still need to use `LoadToStore` here and check the result because a concurrent goroutine could have already created the entry between the `Load` and creation of the `cleanUpEntry`.

Reply:
Good point 👍